pyfolio/capacity.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import division
+import pandas as pd
+import numpy as np
+from . import pos
+from . import timeseries
+
+DISPLAY_UNIT = 1000000
+
+
+def get_assets_dollar_volume(positions, market_data,
+                             only_held_days=True, last_n_days=None):
+    """
+    Computes descriptive statsitics for the daily dollar
+    volume for traded names over the backtest dates.
+
+    Parameters
+    ----------
+    positions: pd.DataFrame
+        Contains daily position values including cash
+        - See full explanation in tears.create_full_tear_sheet
+    market_data : pd.Panel
+        Contains OHLCV DataFrames for the tickers in the passed
+        positions DataFrame
+    only_held_days : boolean
+        Only compute daily dollar volume statistics on days when
+        a name was held in the portfolio.
+    last_n_days : integer, optional
+        Compute max position allocation and dollar volume for only
+        the last N days of the backtest
+
+    Returns
+    -------
+    vol_analysis : pd.DataFrame
+        Index of traded tickers. Columns for 10th percentile, 90th percentile,
+        and mean daily traded volume, as well as max position concentration.
+    """
+    DV = market_data['volume'] * market_data['price']
+
+    positions_alloc = pos.get_percent_alloc(positions)
+    positions_alloc = positions_alloc.drop('cash', axis=1)
+
+    if last_n_days is not None:
+        positions_alloc = positions_alloc.iloc[-last_n_days:]
+        DV = DV.iloc[-last_n_days:]
+
+    if only_held_days:
+        for name, ts in positions_alloc.iteritems():
+            DV[name] = DV.loc[ts[ts != 0].index, name]
+
+    max_exposure_per_ticker = abs(positions_alloc).max()
+
+    vol_analysis = pd.DataFrame()
+    vol_analysis['algo_max_exposure'] = max_exposure_per_ticker
+    vol_analysis.loc[:, 'avg_daily_dv_$mm'] = np.round(
+        DV.mean() / DISPLAY_UNIT, 2)
+    vol_analysis['10th_%_daily_dv_$mm'] = np.round(DV.apply(
+        lambda x: np.nanpercentile(x, 10)) / DISPLAY_UNIT, 2)
+    vol_analysis['90th_%_daily_dv_$mm'] = np.round(DV.apply(
+        lambda x: np.nanpercentile(x, 90)) / DISPLAY_UNIT, 2)
```

Comment:
This whole block would look much better if the round and display unit conversion would be done at the end over the columns you want.